lib/src/global.dart

```diff
@@ -17,3 +17,72 @@ Future<void> installOfflineMapTiles(String tilesDb) async {
     },
   );
 }
+
+Future<List<OfflineRegion>> downloadListOfRegions({String accessToken}) async {
+  String regionsJson = await _globalChannel.invokeMethod(
+    'downloadListOfRegions',
+    <String, dynamic>{
+      'accessToken': accessToken,
+    },
+  );
+  Iterable regions = json.decode(regionsJson);
+  return regions.map((region) => OfflineRegion.fromJson(region)).toList();
+}
+
+Future<dynamic> deleteOfflineRegion(int id, {String accessToken}) =>
+    _globalChannel.invokeMethod(
+      'deleteOfflineRegion',
+      <String, dynamic>{
+        'id': id,
+        'accessToken': accessToken,
+      },
+    );
+
+Future<dynamic> downloadOfflineRegion(OfflineRegion region,
+        {String accessToken}) =>
+    _globalChannel.invokeMethod(
+      'downloadOfflineRegion',
+      json.encode(
+        region._toJson()..putIfAbsent('accessToken', () => accessToken),
+      ),
+    );
+
+void downloadOfflineRegionStream(
+  OfflineRegion region,
+  Function(DownloadRegionStatus event) onEvent, {
+  String accessToken,
+}) async {
+  downloadOfflineRegion(region, accessToken: accessToken);
+  String channelName = 'downloadOfflineRegion_${region.id}';
+  EventChannel(channelName).receiveBroadcastStream().handleError((error) {
+    if (error is PlatformException) {
+      onEvent(Error(error));
+      return Error(error);
+    }
+    var unknownError = Error(
+      PlatformException(
+        code: 'UnknowException',
+        message:
+            'This error is unhandled by plugin. Please contact us if needed.',
+        details: error,
+      ),
+    );
+    onEvent(unknownError);
+    return unknownError;
+  }).listen((data) {
+    final Map<String, dynamic> jsonData = json.decode(data);
+    DownloadRegionStatus status;
+    switch (jsonData['status']) {
+      case 'start':
+        status = InProgress(0.0);
+        break;
+      case 'progress':
+        status = InProgress(jsonData['progress'] ?? 0.0);
```

Comment:
It seems that jsonData['progress'] may be a integer in iOS. I did not debug it in the detail.
```
[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: type 'int' is not a subtype of type 'double'
#0      downloadOfflineRegionStream.<anonymous closure> (package:mapbox_gl/src/global.dart:80:50)
#1      _rootRunUnary (dart:async/zone.dart:1192:38)
#2      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#3      _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)
#4      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#5      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:266:7)
#6      _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:134:11)
#7      _ForwardingStream._handleData (dart:async/stream_pipe.dart:100:10)
#8      _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:166:13)
#9      _rootRunUnary (dart:async/zone.dart:1192:38)
#10     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#11     _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)
#12     _BufferingStreamSubscription._sendData (dart:asyn<…>
```